File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l = 0
        if s== "":
            return True
        for r in range(len(t)):
            if l >= len(s):
                break
            if t[r] == s[l]:
                l += 1
        if l == len(s):
            return True
        else:
            return False
            
            
        

File: 0392-is-subsequence/0392-is-subsequence.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let (mut i, mut j) = (0, 0);
        let vec_s = s.chars().collect::<Vec<char>>();
        let vec_t = t.chars().collect::<Vec<char>>();
        while i < s.len() && j < t.len(){
            if vec_s[i] == vec_t[j]{
                i += 1;
            }
            j += 1;
        }
        
        if i >= s.len(){
            true
        }else{
            false
        }
    }
}